protocol-units/da-sequencer/node/src/tests/client.rs

```diff
@@ -29,46 +38,46 @@ async fn test_client_should_successfully_write_batch() {
 	//need to wait the server is started before connecting
 	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(300)).await;
 
-	let tx = Transaction::test_only_new(b"test data".to_vec(), 1, 123);
-
-	let txs = FullNodeTxs::new(vec![tx]);
+	let connection_string = format!("http://127.0.0.1:{}", grpc_address.port());
+	let mut client = DaSequencerClient::try_connect(&connection_string)
+		.await
+		.expect("gRPC client connection failed.");
 
-	//sign batch
-	let batch_bytes = bcs::to_bytes(&txs).expect("Serialization failed");
-	let signature = sign_batch(&batch_bytes, &signing_key);
+	mock_write_new_batch(&mut client, &signing_key, verifying_key).await;
 
-	// Serialize full node batch into raw bytes
-	let serialized =
-		serialize_full_node_batch(verifying_key, signature.clone(), batch_bytes.clone());
-
-	//send the bacth using the grpc client
+	//verify the block produced contains the batch.
+	//Register to block stream
 	let connection_string = format!("http://127.0.0.1:{}", grpc_address.port());
 	let mut client = DaSequencerClient::try_connect(&connection_string)
 		.await
 		.expect("gRPC client connection failed.");
 
-	let request = BatchWriteRequest { data: serialized };
-	let res = client.batch_write(request).await.expect("Failed to send the batch.");
-	tracing::info!("{res:?}",);
-	assert!(res.answer);
-
-	//TODO verify the block produced contains the batch.
-	// Wait the implementation of the stream of block.
+	let mut block_stream = client
+		.stream_read_from_height(StreamReadFromHeightRequest { height: 0 })
+		.await
+		.expect("Failed to register to block stream");
 
 	//wait at least one block production
 	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+	mock_wait_and_get_next_block(&mut block_stream, 1).await;
+
+	grpc_jh.abort();
+	loop_jh.abort();
 }
 
 #[tokio::test]
-async fn test_batch_write_should_fail_with_wrong_sig() {
+async fn test_write_batch_gprc_main_loop_happy_path_unhappy_path() {
```

Comment:
```suggestion
async fn test_write_batch_gprc_main_loop_happy_path_unhappy_path() {
```

Improve the test name, let's avoid unhappy or happy path in unit test names. Express what you are testing for example 
`test_grpc_loop_should_stream_block_heights_in_order` for example. 

`test_apples_should_be_red_not_black`

Fully express what is being tested, or what is exepected to fail. It makes it much easier for the reader to understand the intention of your code.

Reply:
I've renamed the test name.